src/main/resources/api/service_openapi.yaml

```diff
@@ -72,6 +72,27 @@ paths:
         500:
           description: Creation request error
           $ref: '#/components/responses/ErrorResponse'
+    get:
+      description: |
+        Return tracked resources matching the request filters
+      operationId: getResources
+      tags:
+        - janitor
+      parameters:
+        - name: cloudResourceUid
+          in: query
```

Comment:
Not sure what would be the best practice for defining a get for an object. In your solution, we treat the whole things as a string, not a query. And the rest api would be 

`get https://janitor/api/resource? cloudResourceUid=[a super long string]. `
Do we want to go with this solution or we want break down the cloudResourceUid and make it more like an 
"object". something like: 
`get https://janitor/api/resource? resourceType=[type]&projectName=[]&bucketName=[]`?
I don't know which is more preferred.

Reply:
I'm pretty strongly opposed to having another serialization format for a "cloud resource uid." I'm open to other singular formats. We have the use case of using it within a request body as a json object, and now in a get params (or path).

Doing this seemed not too bad. The string doesn't have to be too long if it's serialized to not include nulls, probably doesn't look more different than the individual params you're suggesting. It also gives us a single object to describe the "cloud resource uid" in the library and client apis rather than an ever expanding list of parameters.